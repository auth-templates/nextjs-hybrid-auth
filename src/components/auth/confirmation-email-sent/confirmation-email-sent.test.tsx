import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ConfirmationEmailSent from './confirmation-email-sent';

describe("ConfirmationEmailSent", () => {
    it('should contain all elements', async () => {
        render(
            <MemoryRouter>
                <ConfirmationEmailSent email={'account@mail.com'} />
            </MemoryRouter>
        );

        expect(screen.getByText(/A confirmation email has been sent to/)).toBeInTheDocument(); 
        expect(screen.getByText(/account@mail.com/)).toBeInTheDocument();
        expect(screen.getByText(/and it should arrive soon. Please check your inbox./)).toBeInTheDocument();
    });

    it("renders correctly", () => {
        const tree = renderer.create(
            <MemoryRouter>
                <ConfirmationEmailSent email={'account@mail.com'} /> 
            </MemoryRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});