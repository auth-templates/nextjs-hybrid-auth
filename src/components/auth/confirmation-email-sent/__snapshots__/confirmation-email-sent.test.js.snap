// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationEmailSent renders correctly 1`] = `
<div
  className="confirmationEmailSent"
>
  A confirmation email has been sent to
  <div
    className="email"
  >
     
    account@mail.com
     
  </div>
  and it should arrive soon. Please check your inbox.
</div>
`;
